### ISSUE TITLE
require-return-type shouldn't complain about constructors

### ISSUE THREAD

**Issue by joewalker** [gajus/eslint-plugin-flowtype#281]
I'm being explicit about returning undefined with `:void`, but when I enforce this with an `annotateUndefined: always` setting, it catches constructors as violations...

``` js
export class Demo {
  constructor() { } // Must annotate undefined return type. (flowtype/require-return-type)
}
```

It's doesn't look possible to work around this with `excludeMatching`:

``` js
"flowtype/require-return-type": [
  "warn",
  "always",
  {
    "annotateUndefined": "always",
    "excludeMatching": [ "constructor" ] // doesn't work
  }
],
```

Could constructors be excluded from `annotateUndefined: always` please?

**Comment by deecewan** [comment]
I'm going to have a crack at this - should be a pretty easy fix I think.

**Comment by joewalker** [comment]
@deecewan I tested the fix out and it works. Fantastic! Thanks.

> edit: @joewalker I made the fix you asked for, but I don't necessarily think it's the correct one. Why should constructors not have an explicit return type?

I guess `class Gak { constructor() { return new Date(); } }` is legal, so yes - I think your fix is the best one.
